src/OrchardCore/OrchardCore.Users.Core/Services/SuperAdminClaimsProviders.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Security;
+using OrchardCore.Security.Permissions;
+using OrchardCore.Settings;
+using OrchardCore.Users.Models;
+
+namespace OrchardCore.Users.Services;
+
+public class SuperAdminClaimsProviders : IUserClaimsProvider
+{
+    private readonly IServiceProvider _serviceProvider;
+    private ISiteService _siteService;
+    private IEnumerable<IPermissionProvider> _permissionProviders;
+    private IRoleStore<IRole> _roleStore;
+
+    public SuperAdminClaimsProviders(IServiceProvider serviceProvider)
+    {
+        _serviceProvider = serviceProvider;
+    }
+
+    public async Task GenerateAsync(IUser user, ClaimsIdentity claims)
+    {
+        _roleStore = _serviceProvider.GetService<IRoleStore<IRole>>();
+        _siteService ??= _serviceProvider.GetService<ISiteService>();
+        _permissionProviders ??= _serviceProvider.GetServices<IPermissionProvider>();
+
+        // NullRoleStore is only registered when Roles features is disabled.
+        // So we load all available claims for the super user only to grant them to all available permissions.
+        // These provider should only apply when NullRoleStore is registered.
+        if (
+            _roleStore is not NullRoleStore
+            || user is not User su
+            || _siteService == null
+            || _permissionProviders == null
+            || !_permissionProviders.Any())
+        {
+            return;
+        }
+
+        var site = await _siteService.GetSiteSettingsAsync();
+
+        if (site.SuperUser != null && String.Equals(site.SuperUser, su.UserId))
+        {
+            var permissionCollections = await Task.WhenAll(_permissionProviders.Select(x => x.GetPermissionsAsync()));
+
+            var permissionClaims = permissionCollections.SelectMany(permissionCollection => permissionCollection)
+                .Select(permission => new Claim(Permission.ClaimType, permission.Name));
+
+            claims.AddClaims(permissionClaims);
```

Comment:
Not a complete review, just a first comment.

Yes the site service document is cached by the DocumentManager so that's okay.

Maybe replace `x => x.` by `provider => provider.`

I would not use `Task.WhenAll()` as we don't know what `GetPermissionsAsync()` may do.

Reply:
actually this class need to be removed. we already have permission handler that grant all permission to super admin.

Reply:
Okay then

Reply:
removed this class. The SuperUserHandler already grants the super everything https://github.com/OrchardCMS/OrchardCore/blob/339d74b251b53393dd510f8ad3027aa17d78500d/src/OrchardCore.Modules/OrchardCore.Settings/Services/SuperUserHandler.cs